fs/unstable_read_text_file.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+import { mapError } from "./_map_error.ts";
+import type { ReadFileOptions } from "./unstable_types.ts";
+import { isDeno } from "./_utils.ts";
+import { readFile, readFileSync } from "./unstable_read_file.ts";
+
+/**
+ * @example Usage
+ * ```ts
+ * import { readTextFile } from "@std/fs/unstable-read-text-file";
+ * const content = await readTextFile("README.md"); // full content of README.md
+ * console.log(content);
+ * ```
+ *
+ * @tags allow-read
+ *
+ * @param path The path of the symbolic link.
+ * @param options Options when reading a file. See {@linkcode ReadFileOptions}.
+ * @returns A promise that resolves to string of the file content.
+ */
+export async function readTextFile(
+  path: string | URL,
+  options?: ReadFileOptions,
+): Promise<string> {
+  if (isDeno) {
+    return Deno.readTextFile(path, { ...options });
+  } else {
+    try {
+      const decoder = new TextDecoder("utf-8");
+      const data = await readFile(path, options);
+      return decoder.decode(data);
+    } catch (error) {
+      throw mapError(error);
+    }
+  }
+}
+
+/**
+ * @example Usage
```

Comment:
Ditto. Can you also copy the main description of API from Deno API docs? https://docs.deno.com/api/deno/~/Deno.readTextFileSync

Reply:
Updated.